### PR TITLE
Implement grid mode clip edit & launch - feedback appreciated. Implements #1909

### PR DESCRIPTION
**Pull request by RJ-Eckie**
This implements a way to edit and launch a clip in grid mode with one action.

### Changes to edit mode:
Edit mode is now either launching or non-launching:
- Blue: non-launching (unchanged)
- Cyan: identical to blue, but whenever a clip is entered that clip is also launched (quantized, or immediately when holding shift)

Launching/non-launching mode can be toggled by pressing green+blue at the same time
~~Launching is quantized. (Immediate launching with shift+cyan to be implemented.)~~

### Changes to copying in grid mode:
Copying now has three variants:
- Launch mode: after copying, remain in grid view (unchanged)
- Edit mode, non-launching: after copying, enters newly created clip
- Edit mode, launching: after copying, enters newly created clip _and_ launches it

--

This addresses all complaints in #1909 and even goes a step further. Especially when copying + launching it feels really smooth! Interested to hear what other people think.

### CODE REVIEW

**Comment by RJ-Eckie** [comment]
Ready for review as far as I'm concerned. Happy to make changes if there's feedback.

File: src/deluge/gui/views/session_view.cpp [hunk 1]
```diff
@@ -3188,7 +3190,7 @@ void SessionView::gridRenderActionModes(int32_t y, RGB image[][kDisplayWidth + k
 	}
 	case GridMode::BLUE: {
```
Comment (m-m-adams) [inline src/deluge/gui/views/session_view.cpp:3191]: Cyan should be it's own mode

**Comment by m-m-adams** [comment]
Are you still working on this? We've generally found that switching modes is not popular and are aiming to make it as unnecessary as possible, to do that it needs to be its own mode so it can tie into the momentary options

**Comment by m-m-adams** [comment]
Feel free to reopen if you start working on it again

**Pull request closed without merging**